src/associations/belongs-to-many/belongs-to-many-association.ts

```diff
@@ -26,7 +26,7 @@ export class BelongsToManyAssociation extends BaseAssociation {
     const associatedClass = this.getAssociatedClass();
     const throughOptions = this.getThroughOptions(sequelize);
 
-    const throughModel = typeof throughOptions === 'object' ? throughOptions.model : undefined;
+    const throughModel = typeof throughOptions === 'object' && typeof throughOptions.model !== "string" ? throughOptions.model : undefined;
```

Comment:
Can you explain why this change is necessary?

Reply:
Because `throughOptions.model` has type `string | typeof Model`. The subsequent code isn't designed to handle the string case. Since my goals were to start with the minimal set of changes to make the code compatible with Sequelize 6 beta.6, this was the smallest fix to make the compiler and test suite happy.